<?xml version="1.0"?>

<launch>
  <!-- 0-raw, 1-pos, 2-trj -->
  <param name="flight_mode" value="0" type="int" />
  <param name="joy_num" value="1" type="int" />
  <arg name="joy_dev0" default="/dev/input/js0" />
  <!--include file="$(find easyfly)/launch/swarm_cf.launch" /-->
  <arg name="g_vehicle_num" default="2"/>
  <!--arg name="g_joy_num" default="2"/-->
  <arg name="tf_prefix" default="" />
  <arg name="roll_trim" default="0" />
  <arg name="pitch_trim" default="0" />
  <arg name="enable_logging" default="True" />
  <arg name="use_ros_time" default="True" />
  <arg name="enable_logging_imu" default="True" />
  <arg name="enable_logging_temperature" default="False" />
  <arg name="enable_logging_magnetic_field" default="True" />
  <arg name="enable_logging_pressure" default="False" />
  <arg name="enable_logging_battery" default="False" />
  <param name="enable_logging" value="$(arg enable_logging)" />
  <param name="use_ros_time" value="$(arg use_ros_time)" />
  <param name="enable_logging_imu" value="$(arg enable_logging_imu)" />
  <param name="enable_logging_temperature" value="$(arg enable_logging_temperature)" />
  <param name="enable_logging_magnetic_field" value="$(arg enable_logging_magnetic_field)" />
  <param name="enable_logging_pressure" value="$(arg enable_logging_pressure)" />
  <param name="enable_logging_battery" value="$(arg enable_logging_battery)" />

  <group ns="joygroup0">
    <node pkg="joy" name="joy_node0" type="joy_node" >
      <param name="dev" value="$(arg joy_dev0)" />
    </node>
  </group>

  <group ns="vehicle0">
      <arg name="uri" default="radio://0/0/2M/0xE7E7E7E7E7"/>
      <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add0" output="screen">
          <param name="group_index" value="0"/>
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
  </group>

  <group ns="vehicle1">
      <arg name="uri" default="radio://0/110/250K/0xE7E7E7E7E7"/>
      <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add1" output="screen">
        <param name="uri" value="$(arg uri)" type="string" />
        <param name="group_index" value="1"/>
        <param name="tf_prefix" value="$(arg tf_prefix)" />
        <param name="roll_trim" value="$(arg roll_trim)" />
        <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
  </group>

<node pkg="easyfly" name="commander" type="commander" output="screen"/>
<node pkg="easyfly" type="swarm_linker" name="swarm_linker" output="screen" />
<node pkg="easyfly" type="swarm_controller" name="swarm_controller" output="screen" />

</launch>
